From epasheva@sierrawireless.com  Tue Jun  2 13:27:24 2009
From: Elina Pasheva <epasheva@sierrawireless.com>
Date: Mon, 1 Jun 2009 11:01:14 -0700
Subject: USB: serial: adding sierra_stop_rx_urbs() function
To: <gregkh@suse.de>
Cc: <epasheva@sierrawireless.com>, <rfiler@sierrawireless.com>
Message-ID: <1243879274.6383.4.camel@Linuxdev3>


From: Elina Pasheva <epasheva@sierrawireless.com>

- Added function sierra_stop_rx_urbs() that takes care of the release of
  receive and interrupt urbs. This function is to be called by
  sierra_close() whenever an interface is de-activated.

Signed-off-by: Elina Pasheva <epasheva@sierrawireless.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/usb/serial/sierra.c |   11 +++++++++++
 1 file changed, 11 insertions(+)

--- a/drivers/usb/serial/sierra.c
+++ b/drivers/usb/serial/sierra.c
@@ -571,6 +571,17 @@ static int sierra_write_room(struct tty_
 	return 2048;
 }
 
+static void sierra_stop_rx_urbs(struct usb_serial_port *port)
+{
+	int i;
+	struct sierra_port_private *portdata = usb_get_serial_port_data(port);
+
+	for (i = 0; i < ARRAY_SIZE(portdata->in_urbs); i++)
+		usb_kill_urb(portdata->in_urbs[i]);
+
+	usb_kill_urb(port->interrupt_in_urb);
+}
+
 static int sierra_open(struct tty_struct *tty,
 			struct usb_serial_port *port, struct file *filp)
 {
